crates/pathfinder/src/merkle_tree.rs

```diff
@@ -0,0 +1,1089 @@
+//! Starknet utilises a custom Binary Merkle-Patrica Tree to store and organise its state.
```

Comment:
Overall no comments but great authoring and nice! My main point is probably the use of RefCell and implied (mut_)borrow and dereferences that obfuscate reading it a bit, but very nice comments, tests etc.

Finally a few functions are a bit long (like `set(..)`) but I don't see much gain in splitting them up (maybe a bit readability, but they are well-commented already), so not proposing any change.

Reply:
Thanks for the very thorough review!